graphene_parser.py

```diff
@@ -740,151 +445,135 @@ def __init__(
         self._scope = scope
         self._generic_mapping = generic_mapping
 
-    def visit(self, tree: Tree | Token) -> FlattenedExpression:
-        result = super().visit(tree)
-        assert isinstance(result, FlattenedExpression)
-        return result
-
-    @inline_arguments
-    def expression(self, expr_tree: Tree) -> FlattenedExpression:
-        return self.visit(expr_tree)
+    def parse_expr(self, expr: lp.Expression) -> FlattenedExpression:
+        return self.parse(expr, FlattenedExpression)
 
-    def UNSIGNED_HEX_CONSTANT(self, token: Token) -> FlattenedExpression:
-        const_expr = cg.ConstantExpression(cg.UIntType(), token.value)
+    def HexConstant(self, node: lp.HexConstant) -> FlattenedExpression:
+        const_expr = cg.ConstantExpression(cg.UIntType(), node.value)
         return FlattenedExpression([const_expr])
 
-    def NUMERIC_GENERIC_IDENTIFIER(self, token: Token) -> FlattenedExpression:
-        argument = cg.GenericArgument(token.value, True)
+    def GenericIdentifierConstant(
+        self, node: lp.GenericIdentifierConstant
+    ) -> FlattenedExpression:
+        argument = cg.GenericArgument(node.value, True)
         if argument not in self._generic_mapping.mapping:
-            raise FailedLookupError("numeric generic", f"[{token.value}, ...]")
+            raise FailedLookupError("numeric generic", f"[{node.value}, ...]")
 
         mapped_value = self._generic_mapping.mapping[argument]
         assert isinstance(mapped_value, int)  # TODO: user facing error
         const_expr = cg.ConstantExpression(cg.IntType(), str(mapped_value))
         return FlattenedExpression([const_expr])
 
-    def SIGNED_INT(self, token: Token) -> FlattenedExpression:
-        const_expr = cg.ConstantExpression(cg.IntType(), token.value)
+    def IntConstant(self, node: lp.IntConstant) -> FlattenedExpression:
+        # TODO: the parser has already decoded this, why are we undoing it?
```

Comment:
The idea was "why bother if we're just gonna forward it to llvm", but we'll need the actual value soon enough---when I dig up that automatic constant-sizing commit

Reply:
Ooo, yes I was missing this, that would be cool.

I gave up on the Unicode library partially because it was too painful to work without automatic const-sizing + char literals (also because I released the parser didn't actually need it :smile:).

Reply:
Fair enough. We technically do need a unicode library to properly do json though :/